### PR TITLE
yammer metrics 2.0.0 support

### PR DESCRIPTION
**Pull request by lganzzzo**
Added support for yammer.metrics v.2.0.0

### CODE REVIEW

**Comment by cep21** [comment]
@lganzzzo Can you check the travis build and make it build on jdk8?  @mdubbyap heads up here

**Comment by cep21** [comment]
It's probably javadoc stuff

**Comment by lganzzzo** [comment]
done

File: signalfx-yammer/auth.properties [hunk 1]
```diff
@@ -0,0 +1,2 @@
+auth=xznqOC3qDvPytQeUdivlMQ
```
Comment (mdubbyap) [inline signalfx-yammer/auth.properties]: we do not want to check in an auth token

File: README.md [hunk 2]
```diff
@@ -136,6 +186,11 @@ use code like this.
 
 ```
 
+```Yammer Metrics 2.0.x
+
+NOT SUPPORTED;
```
Comment (mdubbyap) [inline README.md]: For my own education, why is this not supported?

File: signalfx-yammer/pom.xml [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+  <scm>
+    <connection>scm:git:git@github.com:signalfx/signalfx-clients.git</connection>
+    <developerConnection>scm:git:git@github.com:signalfx/signalfx-clients.git</developerConnection>
+    <url>git@github.com:signalfx/signalfx-clients.git</url>
+  </scm>
+
+  <developers>
+    <developer>
+      <id>jack</id>
```
Comment (mdubbyap) [inline signalfx-yammer/pom.xml]: Can we remove jack? just have organization stuff.  If you need an email put info@signalfx.com

File: signalfx-yammer/src/main/java/com/signalfx/codahale/reporter/AggregateMetricSenderSessionWrapper.java [hunk 4]
```diff
@@ -0,0 +1,273 @@
...
+
+class AggregateMetricSenderSessionWrapper implements Closeable {
+    private final AggregateMetricSender.Session metricSenderSession;
+    private final Set<SignalFxReporter.MetricDetails> detailsToAdd;
+    private final MetricMetadata metricMetadata;
+    private final String defaultSourceName;
+    private final String sourceDimension;
+    private final boolean injectCurrentTimestamp;
+
+    /*AggregateMetricSenderSessionWrapper(
```
Comment (mdubbyap) [inline signalfx-yammer/src/main/java/com/signalfx/codahale/reporter/AggregateMetricSenderSessionWrapper.java]: Any reason this is commented out?

File: signalfx-yammer/src/main/java/com/signalfx/codahale/reporter/AggregateMetricSenderSessionWrapper.java [hunk 5]
```diff
@@ -0,0 +1,273 @@
...
+        }
+    }
+
+    // These three called from report
+    void addTimer(MetricName key, Timer value) {
+        addMetered(key, value);
+        addSampling(key, value);
+    }
+
+    void addHistogram(MetricName baseName,
```
Comment (mdubbyap) [inline signalfx-yammer/src/main/java/com/signalfx/codahale/reporter/AggregateMetricSenderSessionWrapper.java]: Feel like this file looks almost exactly like https://github.com/signalfx/signalfx-java/blob/master/signalfx-codahale/src/main/java/com/signalfx/codahale/reporter/AggregateMetricSenderSessionWrapper.java

Can this not be refactored out to be common to both?

File: signalfx-yammer/src/main/java/com/signalfx/codahale/reporter/MetricMetadata.java [hunk 6]
```diff
@@ -0,0 +1,74 @@
+package com.signalfx.codahale.reporter;
```
Comment (mdubbyap) [inline signalfx-yammer/src/main/java/com/signalfx/codahale/reporter/MetricMetadata.java]: This file looks nearly if not identical to https://github.com/signalfx/signalfx-java/blob/master/signalfx-codahale/src/main/java/com/signalfx/codahale/reporter/MetricMetadata.java can you refactor this please?

File: signalfx-yammer/src/main/java/com/signalfx/codahale/reporter/MetricMetadataImpl.java [hunk 7]
```diff
@@ -0,0 +1,141 @@
+package com.signalfx.codahale.reporter;
```
Comment (mdubbyap) [inline signalfx-yammer/src/main/java/com/signalfx/codahale/reporter/MetricMetadataImpl.java]: Also appears to be identical to https://github.com/signalfx/signalfx-java/blob/master/signalfx-codahale/src/main/java/com/signalfx/codahale/reporter/MetricMetadataImpl.java

File: signalfx-yammer/src/main/java/com/signalfx/codahale/reporter/SfUtil.java [hunk 8]
```diff
@@ -0,0 +1,42 @@
+package com.signalfx.codahale.reporter;
+
+import com.yammer.metrics.core.Gauge;
```
Comment (mdubbyap) [inline signalfx-yammer/src/main/java/com/signalfx/codahale/reporter/SfUtil.java]: More duplicate of https://github.com/signalfx/signalfx-java/blob/master/signalfx-codahale/src/main/java/com/signalfx/codahale/reporter/SfUtil.java

**Comment by mdubbyap** [comment]
Ok, so i've stopped reviewing.  Can you refactor out all the duplicate code first, then i'll take another look.  Thanks.

File: signalfx-yammer/src/main/java/com/signalfx/codahale/reporter/AggregateMetricSenderSessionWrapper.java (hunk 4 above)
Comment (lganzzzo) [inline signalfx-yammer/src/main/java/com/signalfx/codahale/reporter/AggregateMetricSenderSessionWrapper.java]: This constructor never used. Deleted

File: signalfx-yammer/src/main/java/com/signalfx/codahale/reporter/AggregateMetricSenderSessionWrapper.java (hunk 5 above)
Comment (lganzzzo) [inline signalfx-yammer/src/main/java/com/signalfx/codahale/reporter/AggregateMetricSenderSessionWrapper.java]: No. They using different libraries... one operates with String, other with MetricName

File: signalfx-yammer/src/main/java/com/signalfx/codahale/reporter/MetricMetadata.java (hunk 6 above)
Comment (lganzzzo) [inline signalfx-yammer/src/main/java/com/signalfx/codahale/reporter/MetricMetadata.java]: No.
- public <M extends Metric> BuilderTagger<M> forBuilder(MetricBuilder<M> metricBuilder);
- public interface BuilderTagger<M extends Metric> extends TaggerBase<M, BuilderTagger<M>> ...

Removed.

Different interfaces

File: signalfx-yammer/src/main/java/com/signalfx/codahale/reporter/MetricMetadataImpl.java (hunk 7 above)
Comment (lganzzzo) [inline signalfx-yammer/src/main/java/com/signalfx/codahale/reporter/MetricMetadataImpl.java]: What do you mean same? Implements different interface, uses different library

File: signalfx-yammer/src/main/java/com/signalfx/codahale/reporter/SfUtil.java (hunk 8 above)
Comment (lganzzzo) [inline signalfx-yammer/src/main/java/com/signalfx/codahale/reporter/SfUtil.java]: No, method removeMetrics not supported, uses different library.
I can't see duplicated

**Comment by lganzzzo** [comment]
Please review carefully, there is no duplicates

File: README.md (hunk 2 above)
Comment (lganzzzo) [inline README.md]: There is not method MetricsRegistry::register(...)
So we can not offer user helper functionality to track duplicates in created Metrics instances

**Pull request merged**